File: frontend/src/types/analysis.ts
/**
 * AI 분석 관련 타입 정의 - Bankability 중심
 */

// 시뮬레이션 컨텍스트 - API에 전달
export interface SimulationContext {
  input_summary: {
    capacity_mw: number;
    efficiency: number;
    capex_billion: number;
    electricity_source: string;
    h2_price: number;
    discount_rate: number;
    project_lifetime: number;
    debt_ratio?: number;
    interest_rate?: number;
    loan_tenor?: number;
  };
  kpi_summary: {
    npv: {
      p50_billion: number;
      p90_billion: number;
      p99_billion: number;
    };
    irr: {
      p50: number;
      p90: number;
      p99: number;
    };
    dscr: {
      min: number;
      avg: number;
    };
    payback_years: number;
    var95_billion: number;
    lcoh: number;
    annual_h2_production: number;
  };
  financing_summary?: {
    debt_ratio: number;
    interest_rate: number;
    loan_tenor: number;
    annual_debt_service_billion: number;
  };
  sensitivity_summary?: Array<{
    variable: string;
    low_change_pct: number;
    high_change_pct: number;
  }>;
  risk_waterfall_summary?: Array<{
    factor: string;
    impact_billion: number;
  }>;
  cashflow_summary?: {
    total_investment_billion: number;
    avg_annual_revenue_billion: number;
    avg_annual_opex_billion: number;
    debt_payoff_year: number;
  };
  incentives_summary?: {
    itc_enabled: boolean;
    itc_rate: number;
    itc_amount_billion: number;
    ptc_enabled: boolean;
    ptc_amount_per_kg: number;
    ptc_duration_years: number;
    capex_subsidy_billion: number;
    operating_subsidy_per_kg: number;
    operating_subsidy_duration_years: number;
    carbon_credit_enabled: boolean;
    carbon_credit_price: number;
    clean_h2_certification_enabled: boolean;
    clean_h2_premium: number;
    total_capex_reduction_billion: number;
    effective_capex_billion: number;
  };
}

// Bankability 점수
export interface BankabilityScore {
  grade: 'A' | 'B' | 'C' | 'D';
  score: number;
  summary: string;
}

// DSCR 분석
export interface DSCRAnalysis {
  assessment: string;
  covenant_headroom: string;
  stress_resilience: string;
}

// 핵심 리스크
export interface KeyRisk {
  risk: string;
  severity: '높음' | '중간' | '낮음';
  impact: string;
  mitigation: string;
}

// Bankability 개선 항목
export interface BankabilityImprovement {
  action: string;
  expected_impact: string;
  priority: '필수' | '권장' | '선택';
  implementation: string;
}

// 금융 권고사항
export interface FinancingRecommendations {
  optimal_leverage: string;
  recommended_tenor: string;
  required_reserves: string;
  covenant_structure: string;
}

// 결과 해석 응답 - Bankability 중심
export interface InterpretResponse {
  executive_summary: string;
  bankability_score: BankabilityScore;
  dscr_analysis: DSCRAnalysis;
  key_risks: KeyRisk[];
  bankability_improvements: BankabilityImprovement[];
  financing_recommendations: FinancingRecommendations;
  lender_concerns: string[];
  investment_readiness: string;
}

// 채팅 메시지
export interface ChatMessage {
  id: string;
  role: 'user' | 'assistant';
  content: string;
  timestamp: Date;
  toolResults?: ToolResult[];
}

// Tool 실행 결과
export interface ToolResult {
  tool: string;
  input: Record<string, unknown>;
  result: Record<string, unknown>;
}

// 채팅 응답
export interface ChatResponse {
  message: string;
  tool_results?: ToolResult[];
}

// 시나리오 비교 응답
export interface CompareResponse {
  comparison_summary: string;
  bankability_ranking: Array<{
    rank: number;
    scenario: string;
    dscr_p90: string;
    bankability_grade: string;
    key_strength: string;
    key_weakness: string;
  }>;
  sensitivity_comparison: string;
  optimal_structure: string;
  recommendation: string;
}

// 섹션별 설명 응답
export interface ExplainResponse {
  section: string;
  title: string;
  summary: string;
  explanation: string;
  key_insights: string[];
}

// 섹션 타입
export type SectionType = 'kpi' | 'npv_distribution' | 'sensitivity' | 'waterfall' | 'cashflow' | 'heatmap';

// 추천 질문
export interface SuggestedQuestion {
  id: string;
  text: string;
  category: 'dscr' | 'leverage' | 'risk' | 'structure' | 'incentives';
}

// Bankability 등급 색상 매핑
export const BANKABILITY_GRADE_COLORS: Record<string, { bg: string; text: string; border: string; label: string }> = {
  'A': { bg: 'bg-emerald-50', text: 'text-emerald-700', border: 'border-emerald-200', label: 'Investment Grade' },
  'B': { bg: 'bg-blue-50', text: 'text-blue-700', border: 'border-blue-200', label: 'Acceptable' },
  'C': { bg: 'bg-amber-50', text: 'text-amber-700', border: 'border-amber-200', label: 'Marginal' },
  'D': { bg: 'bg-red-50', text: 'text-red-700', border: 'border-red-200', label: 'Sub-investment' },
};

// 우선순위 색상 매핑
export const PRIORITY_COLORS: Record<string, { bg: string; text: string }> = {
  '필수': { bg: 'bg-red-100', text: 'text-red-700' },
  '권장': { bg: 'bg-amber-100', text: 'text-amber-700' },
  '선택': { bg: 'bg-gray-100', text: 'text-gray-600' },
};

// 심각도 색상 매핑
export const SEVERITY_COLORS: Record<string, { bg: string; text: string }> = {
  '높음': { bg: 'bg-red-100', text: 'text-red-700' },
  '중간': { bg: 'bg-amber-100', text: 'text-amber-700' },
  '낮음': { bg: 'bg-green-100', text: 'text-green-700' },
};

// Bankability 관점 추천 질문 목록
export const SUGGESTED_QUESTIONS: SuggestedQuestion[] = [
  {
    id: 'q1',
    text: 'DSCR을 1.30 이상으로 높이려면 어떤 조치가 필요한가요?',
    category: 'dscr',
  },
  {
    id: 'q2',
    text: '현재 구조에서 최적의 부채비율은 얼마인가요?',
    category: 'leverage',
  },
  {
    id: 'q3',
    text: '전력가격이 20% 상승하면 Covenant 위반 가능성이 있나요?',
    category: 'risk',
  },
  {
    id: 'q4',
    text: '대출기관이 가장 우려할 리스크는 무엇이고 어떻게 완화할 수 있나요?',
    category: 'risk',
  },
  {
    id: 'q5',
    text: '오프테이크 계약 구조를 어떻게 개선해야 금리를 낮출 수 있나요?',
    category: 'structure',
  },
  {
    id: 'q6',
    text: 'DSRA와 MRA 적립금은 얼마나 필요한가요?',
    category: 'structure',
  },
  {
    id: 'q7',
    text: '현재 적용 가능한 세액공제와 보조금 혜택을 분석해 주세요.',
    category: 'incentives',
  },
  {
    id: 'q8',
    text: '청정수소 인증을 받으면 수익성이 얼마나 개선되나요?',
    category: 'incentives',
  },
];

File: backend/app/schemas/analysis.py
"""
AI 분석 관련 스키마 정의 - Bankability 중심
"""
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field


class SimulationContext(BaseModel):
    """시뮬레이션 컨텍스트 - Claude에 전달할 구조화된 데이터"""

    # 입력 설정 요약
    input_summary: Dict[str, Any] = Field(description="입력 설정 요약")

    # KPI 요약
    kpi_summary: Dict[str, Any] = Field(description="핵심 지표 요약")

    # 재무 구조 요약 (선택)
    financing_summary: Optional[Dict[str, Any]] = Field(
        default=None, description="재무 구조 요약"
    )

    # 민감도 분석 요약 (선택)
    sensitivity_summary: Optional[List[Dict[str, Any]]] = Field(
        default=None, description="민감도 분석 요약"
    )

    # 리스크 폭포수 요약 (선택)
    risk_waterfall_summary: Optional[List[Dict[str, Any]]] = Field(
        default=None, description="리스크 폭포수 요약"
    )

    # 현금흐름 요약 (선택)
    cashflow_summary: Optional[Dict[str, Any]] = Field(
        default=None, description="현금흐름 요약"
    )


class InterpretRequest(BaseModel):
    """결과 해석 요청"""
    context: SimulationContext = Field(description="시뮬레이션 컨텍스트")
    language: str = Field(default="ko", description="응답 언어 (ko/en)")


# Bankability 점수
class BankabilityScore(BaseModel):
    """Bankability 점수"""
    grade: str = Field(description="등급 (A/B/C/D)")
    score: int = Field(description="점수 (0-100)")
    summary: str = Field(description="등급 판단 근거")


# DSCR 분석
class DSCRAnalysis(BaseModel):
    """DSCR 분석 결과"""
    assessment: str = Field(description="DSCR 분석 결과")
    covenant_headroom: str = Field(description="Covenant 여유도 분석")
    stress_resilience: str = Field(description="스트레스 시나리오 대응력")


# 리스크 항목
class KeyRisk(BaseModel):
    """핵심 리스크 항목"""
    risk: str = Field(description="리스크명")
    severity: str = Field(description="심각도 (높음/중간/낮음)")
    impact: str = Field(description="영향")
    mitigation: str = Field(description="완화 방안")


# Bankability 개선 항목
class BankabilityImprovement(BaseModel):
    """Bankability 개선 항목"""
    action: str = Field(description="개선 조치")
    expected_impact: str = Field(description="예상 효과")
    priority: str = Field(description="우선순위 (필수/권장/선택)")
    implementation: str = Field(description="실행 방법")


# 금융 권고사항
class FinancingRecommendations(BaseModel):
    """금융 구조 권고사항"""
    optimal_leverage: str = Field(description="권장 부채비율")
    recommended_tenor: str = Field(description="권장 대출기간")
    required_reserves: str = Field(description="필요 적립금 규모")
    covenant_structure: str = Field(description="권장 Covenant 구조")


class InterpretResponse(BaseModel):
    """결과 해석 응답 - Bankability 중심"""
    executive_summary: str = Field(description="Bankability 관점의 핵심 결론")
    bankability_score: BankabilityScore = Field(description="Bankability 점수")
    dscr_analysis: DSCRAnalysis = Field(description="DSCR 분석")
    key_risks: List[KeyRisk] = Field(description="핵심 리스크")
    bankability_improvements: List[BankabilityImprovement] = Field(
        description="Bankability 개선 방안"
    )
    financing_recommendations: FinancingRecommendations = Field(
        description="금융 구조 권고사항"
    )
    lender_concerns: List[str] = Field(description="대출기관 우려사항")
    investment_readiness: str = Field(description="투자유치 준비도")


class ChatMessage(BaseModel):
    """채팅 메시지"""
    role: str = Field(description="메시지 역할 (user/assistant)")
    content: str = Field(description="메시지 내용")


class ChatRequest(BaseModel):
    """채팅 요청"""
    context: SimulationContext = Field(description="시뮬레이션 컨텍스트")
    messages: List[ChatMessage] = Field(description="대화 이력")
    language: str = Field(default="ko", description="응답 언어")


class ChatResponse(BaseModel):
    """채팅 응답"""
    message: str = Field(description="AI 응답 메시지")
    tool_results: Optional[List[Dict[str, Any]]] = Field(
        default=None, description="Tool 실행 결과 (있는 경우)"
    )


# 시나리오 비교 관련
class ScenarioRanking(BaseModel):
    """시나리오 순위"""
    rank: int = Field(description="순위")
    scenario: str = Field(description="시나리오명")
    dscr_p90: str = Field(description="P90 DSCR")
    bankability_grade: str = Field(description="Bankability 등급")
    key_strength: str = Field(description="핵심 강점")
    key_weakness: str = Field(description="핵심 약점")


class CompareRequest(BaseModel):
    """시나리오 비교 요청"""
    scenarios: List[Dict[str, Any]] = Field(description="비교할 시나리오 목록")
    language: str = Field(default="ko", description="응답 언어")


class CompareResponse(BaseModel):
    """시나리오 비교 응답 - Bankability 관점"""
    comparison_summary: str = Field(description="Bankability 관점 비교 요약")
    bankability_ranking: List[ScenarioRanking] = Field(description="Bankability 순위")
    sensitivity_comparison: str = Field(description="시나리오별 리스크 민감도 비교")
    optimal_structure: str = Field(description="최적의 구조 조합 제안")
    recommendation: str = Field(description="최종 권고사항")


# 섹션별 설명 요청/응답
class ExplainRequest(BaseModel):
    """섹션별 설명 요청"""
    section: str = Field(description="설명할 섹션 (kpi/npv_distribution/sensitivity/waterfall/cashflow/heatmap)")
    context: SimulationContext = Field(description="시뮬레이션 컨텍스트")
    language: str = Field(default="ko", description="응답 언어")


class ExplainResponse(BaseModel):
    """섹션별 설명 응답"""
    section: str = Field(description="섹션명")
    title: str = Field(description="설명 제목")
    summary: str = Field(description="핵심 요약 (1-2문장)")
    explanation: str = Field(description="상세 설명 (마크다운)")
    key_insights: List[str] = Field(description="주요 인사이트 (3-5개)")
